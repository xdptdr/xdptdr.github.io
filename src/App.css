body {
	margin:0px;
	padding:0px;
	background:#ABCDEF;
	color:#000000;
	font-family:sans-serif;
}

a {
	text-decoration:none;
	color:#000000;
	text-shadow: 0px 0px 1px #0000FF;
}

.header {
	overflow:hidden;
}
.header:hover {
	overflow:initial;
}

.xdptdr-title {
	color:white;
	font-family:Verdana;
	font-weight:bold;
	letter-spacing:50px;
	transform:rotate(2deg);
	transform-origin:left;
	display:inline-block;
	cursor:pointer;
}

.header:hover .xdptdr-title {
	text-shadow: 0px 0px 20px #FFFFFF;
}

.xdptdr-title:hover {
	color:pink;
}

h1, h2, h3, h4, h5, h6 {
	margin:0px;
	padding:0px;
}

.button.googleButton {
	display: inline-block;
	border: 1px solid #4285f4;
	box-sizing: border-box;
	padding: 0 12px;
	font-weight: bold;
	color: white;
	background: -webkit-linear-gradient(top,#4387fd,#4683ea);
	line-height: 28px;
	border-radius: 2px;
	font-family: arial,sans-serif;
	font-size: 13px;
	cursor:pointer;
}

.button.googleButton.enabled:hover {
	opacity:.8;
}

.button.googleButton.disabled {
	opacity:.2;
	cursor:not-allowed;
}

.button.red {
	background:initial;
	background-color:#c9302c;
	border-color:#ac2925;
}

.menubar {
	background:#273547;
	color:white;
	padding:5px;
	border-radius:2px;
}

.menubar .button {
	display: inline-block;
	border: none;
	box-sizing: border-box;
	padding: 0 12px;
	font-weight: bold;
	color: inherit;
	background: transparent;
	line-height: 28px;
	border-radius: 2px;
	font-family: arial,sans-serif;
	font-size: 13px;
	cursor:pointer;
	font-weight:bold;
}

.menubar .button:focus {
	outline: none;
	opacity:.8;
}

.menubar .button.enabled:hover {
	opacity:.8;
}

.menubar .button.disabled {
	opacity:.2;
	cursor:not-allowed;
}

.menubar .spacing {
	width:10px;
}

.content {
	padding-top:10px;
}