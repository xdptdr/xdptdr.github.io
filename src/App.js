import React from 'react';
import './App.css';
import { connect } from 'react-redux';

import {
  clickButton
} from './actions';


export class App extends React.Component {
	render() {
		return <div>
			<h1>Welcome to XPTDR's React <a href="https://github.com/xdptdr">GitHub</a> page</h1>
			<p>I'm using this page for experimenting with React.</p>
			<p>I got started here with <a href="http://academy.plot.ly/#frontend">the Plotly academy tutorial</a></p>
			<p>I learnt about that tutorial on the <a href="https://facebook.github.io/jest/">Jest</a> project page.</p>
			<p>Now, here's a button.</p>
			<div style={{textAlign:"center"}}><button>I'm a lovely sexy button</button></div>
		</div>;
	}
}

function mapStateToProps(state) {
  return {
        redux: state
    };
}

export default connect(mapStateToProps)(App);