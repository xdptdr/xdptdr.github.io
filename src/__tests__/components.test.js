import React from 'react';
import renderer from 'react-test-renderer';
import { App } from '../App';
import { fromJS } from 'immutable';

describe('components', function() {
  describe('<App />', function() {
it('renders correctly', function() {
 var tree = renderer.create(<App redux={fromJS({})} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});