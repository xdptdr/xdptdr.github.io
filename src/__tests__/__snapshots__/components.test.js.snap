exports[`components <App /> renders correctly 1`] = `
<div>
  <h1>
    Welcome to XPTDR\'s React 
    <a
      href="https://github.com/xdptdr">
      GitHub
    </a>
     page
  </h1>
  <p>
    I\'m using this page for experimenting with React.
  </p>
  <p>
    I got started here with 
    <a
      href="http://academy.plot.ly/#frontend">
      the Plotly academy tutorial
    </a>
  </p>
  <p>
    I learnt about that tutorial on the 
    <a
      href="https://facebook.github.io/jest/">
      Jest
    </a>
     project page.
  </p>
  <p>
    Now, here\'s a button.
  </p>
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }>
    <button>
      I\'m a lovely sexy button
    </button>
  </div>
</div>
`;
